### ISSUE TITLE
triangulation fails

### ISSUE THREAD

**Issue by bazigo76** [DeepLabCut/DeepLabCut#2566]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Good morning,

I'm really in a rush, and I can't find a solution to my problem.
The deeplabcut.triangulation does not work. This error message appears: Shape of passed values is (1200, 18), indices imply (1200, 36). I indeed have my .cvs files obtained by deeplabcut.analyze_videos of my two 2d images which measure 1200 by 18. I suppose that the code cannot bring together the two .cvs files into one (?). What to do ?

### Operating System

Windows 11 Professionnel

### DeepLabCut version

DLC 2.3.9

### DeepLabCut mode

3d

### Device type

NVIDIA GeForce RTX 3060 Laptop GPU

### Relevant log output

```shell
In [8]: deeplabcut.triangulate(config, "C:\\Users\\LILI2\\Docume
   ...: nts\\DLC\\video 3d", videotype='.MP4', filterpredictions
   ...: =True)
List of pairs: [['C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-1-01.MP4', 'C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-2-01.MP4'], ['C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-1-02.MP4', 'C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-2-02.MP4'], ['C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-1-03.MP4', 'C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-2-03.MP4'], ['C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-1-04.MP4', 'C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-2-04.MP4'], ['C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-1-05.MP4', 'C:\\Users\\LILI2\\Documents\\DLC\\video 3d\\camera-2-05.MP4']]
Analyzing video C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01.MP4 using config_file_camera-1
Using snapshot-100 for model C:/Users/LILI2/Documents/DLC/bee2ddroite-lili-2024-04-25\dlc-models\iteration-0\bee2ddroiteApr25-trainset95shuffle1
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\tensorflow\python\keras\engine\base_layer_v1.py:1694: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
Activating extracting of PAFs
2024-04-26 14:13:48.736372: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2024-04-26 14:13:48.737067: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cublas64_11.dll'; dlerror: cublas64_11.dll not found
2024-04-26 14:13:48.737597: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cublasLt64_11.dll'; dlerror: cublasLt64_11.dll not found
2024-04-26 14:13:48.738150: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cufft64_10.dll'; dlerror: cufft64_10.dll not found
2024-04-26 14:13:49.787931: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cusparse64_11.dll'; dlerror: cusparse64_11.dll not found
2024-04-26 14:13:49.788589: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudnn64_8.dll'; dlerror: cudnn64_8.dll not found
2024-04-26 14:13:49.788891: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2024-04-26 14:13:49.789496: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-04-26 14:13:49.828918: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
Starting to analyze %  C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01.MP4
Video already analyzed! C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01DLC_dlcrnetms5_bee2ddroiteApr25shuffle1_100.h5
Using snapshot-100 for model C:/Users/LILI2/Documents/DLC/bee2ddroite-lili-2024-04-25\dlc-models\iteration-0\bee2ddroiteApr25-trainset95shuffle1
Processing...  C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01.MP4
Tracklets already computed C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01DLC_dlcrnetms5_bee2ddroiteApr25shuffle1_100_el.pickle
Set overwrite = True to overwrite.
The tracklets were created (i.e., under the hood deeplabcut.convert_detections2tracklets was run). Now you can 'refine_tracklets' in the GUI, or run 'deeplabcut.stitch_tracklets'.
Processing...  C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01.MP4
100%|█████████████████████████████████████| 1/1 [00:00<?, ?it/s]
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py:927: FutureWarning: Starting with pandas version 3.0 all arguments of to_hdf except for the argument 'path_or_buf' will be keyword-only.
  df.to_hdf(output_name, "tracks", format="table", mode="w")
The videos are analyzed. Time to assemble animals and track 'em...
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
C:\Users\LILI2\Documents\DLC\video 3d camera-1-01 DLC_dlcrnetms5_bee2ddroiteApr25shuffle1_100
Filtering with median model C:\Users\LILI2\Documents\DLC\video 3d\camera-1-01.MP4
Data from camera-1-01 were already filtered. Skipping...
Analyzing video C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01.MP4 using config_file_camera-2
Using snapshot-100 for model C:/Users/LILI2/Documents/DLC/bee2dgauche-lili-2024-04-26\dlc-models\iteration-0\bee2dgaucheApr26-trainset95shuffle1
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\tensorflow\python\keras\engine\base_layer_v1.py:1694: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
Activating extracting of PAFs
Starting to analyze %  C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01.MP4
Video already analyzed! C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01DLC_dlcrnetms5_bee2dgaucheApr26shuffle1_100.h5
Using snapshot-100 for model C:/Users/LILI2/Documents/DLC/bee2dgauche-lili-2024-04-26\dlc-models\iteration-0\bee2dgaucheApr26-trainset95shuffle1
Processing...  C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01.MP4
Tracklets already computed C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01DLC_dlcrnetms5_bee2dgaucheApr26shuffle1_100_el.pickle
Set overwrite = True to overwrite.
The tracklets were created (i.e., under the hood deeplabcut.convert_detections2tracklets was run). Now you can 'refine_tracklets' in the GUI, or run 'deeplabcut.stitch_tracklets'.
Processing...  C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01.MP4
100%|█████████████████████████████████████| 1/1 [00:00<?, ?it/s]
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py:927: FutureWarning: Starting with pandas version 3.0 all arguments of to_hdf except for the argument 'path_or_buf' will be keyword-only.
  df.to_hdf(output_name, "tracks", format="table", mode="w")
The videos are analyzed. Time to assemble animals and track 'em...
 Call 'create_video_with_all_detections' to check multi-animal detection quality before tracking.
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.
C:\Users\LILI2\Documents\DLC\video 3d camera-2-01 DLC_dlcrnetms5_bee2dgaucheApr26shuffle1_100
Filtering with median model C:\Users\LILI2\Documents\DLC\video 3d\camera-2-01.MP4
Data from camera-2-01 were already filtered. Skipping...
Undistorting...
Computing the triangulation...
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\utils\auxiliaryfunctions_3d.py:290: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  for _, group in df.groupby("individuals", axis=1):
C:\Users\LILI2\anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\utils\auxiliaryfunctions_3d.py:290: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  for _, group in df.groupby("individuals", axis=1):
-----------------------------------------------------------------
ValueError                      Traceback (most recent call last)
Cell In[8], line 1
----> 1 deeplabcut.triangulate(config, "C:\\Users\\LILI2\\Documents\\DLC\\video 3d", videotype='.MP4', filterpredictions=True)

File ~\anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\pose_estimation_3d\triangulation.py:468, in triangulate(config, video_path, videotype, filterpredictions, filtertype, gputouse, destfolder, save_as_csv, track_method)
    465 triangulate = triangulate.swapaxes(0, 1).reshape((num_frames, -1))
    467 # Fill up 3D dataframe
--> 468 df_3d = pd.DataFrame(triangulate, columns=columns, index=inds)
    470 df_3d.to_hdf(
    471     str(output_filename + ".h5"),
    472     "df_with_missing",
    473     format="table",
    474     mode="w",
    475 )
    477 # Reorder 2D dataframe in view 2 to match order of view 1

File ~\anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\core\frame.py:827, in DataFrame.__init__(self, data, index, columns, dtype, copy)
    816         mgr = dict_to_mgr(
    817             # error: Item "ndarray" of "Union[ndarray, Series, Index]" has no
    818             # attribute "name"
   (...)
    824             copy=_copy,
    825         )
    826     else:
--> 827         mgr = ndarray_to_mgr(
    828             data,
    829             index,
    830             columns,
    831             dtype=dtype,
    832             copy=copy,
    833             typ=manager,
    834         )
    836 # For data is list-like, or Iterable (will consume into list)
    837 elif is_list_like(data):

File ~\anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\core\internals\construction.py:336, in ndarray_to_mgr(values, index, columns, dtype, copy, typ)
    331 # _prep_ndarraylike ensures that values.ndim == 2 at this point
    332 index, columns = _get_axes(
    333     values.shape[0], values.shape[1], index=index, columns=columns
    334 )
--> 336 _check_values_indices_shape_match(values, index, columns)
    338 if typ == "array":
    339     if issubclass(values.dtype.type, str):

File ~\anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\core\internals\construction.py:420, in _check_values_indices_shape_match(values, index, columns)
    418 passed = values.shape
    419 implied = (len(index), len(columns))
--> 420 raise ValueError(f"Shape of passed values is {passed}, indices imply {implied}")

ValueError: Shape of passed values is (1200, 18), indices imply (1200, 36)
```

### Anything else?

Here my config files for the 3d: 
# Project definitions (do not edit)
Task: lili
scorer: bee3d
date: Apr26

# Project path (change when moving around)
project_path: C:\Users\LILI2\Documents\DLC\lili-bee3d-2024-04-26-3d

# Plotting configuration
skeleton:     # Note that the pairs must be defined, as you want them linked!
- - head
  - center
- - center
  - back
- - flowright
  - flowcenter
- - flowcenter
  - flowleft
skeleton_color: black
pcutoff: 0.6
colormap: rainbow
dotsize: 12
alphaValue: 0.7
markerType: '*'
markerColor: r

# Number of cameras, camera names, path of the config files, shuffle index and trainingsetindex used to analyze videos:
num_cameras: 2
camera_names:
- camera-1
- camera-2
scorername_3d: DLC_3D # Enter the scorer name for the 3D output
trainingsetindex_camera-1: 0
trainingsetindex_camera-2: 0
config_file_camera-1: 
  C:/Users/LILI2/Documents/DLC/bee2ddroite-lili-2024-04-25/config.yaml
config_file_camera-2: 
  C:/Users/LILI2/Documents/DLC/bee2dgauche-lili-2024-04-26/config.yaml
shuffle_camera-1: 1
shuffle_camera-2: 1

And one of my config file 2d:
    # Project definitions (do not edit)
Task: bee2ddroite
scorer: lili
date: Apr25
multianimalproject: true
identity: false

    # Project path (change when moving around)
project_path: C:/Users/LILI2/Documents/DLC/bee2ddroite-lili-2024-04-25

    # Annotation data set configuration (and individual video cropping parameters)
video_sets:
  C:\Users\LILI2\Documents\DLC\bee2ddroite-lili-2024-04-25\videos\camera-1-01.MP4:
    crop: 0, 640, 0, 720
  C:\Users\LILI2\Documents\DLC\bee2ddroite-lili-2024-04-25\videos\camera-1-02.MP4:
    crop: 0, 640, 0, 720
individuals:
- bee
uniquebodyparts:
- flowright
- flowleft
- flowcenter
multianimalbodyparts:
- head
- center
- back
bodyparts: MULTI!

    # Fraction of video to start/stop when extracting frames for labeling/refinement
start: 0
stop: 1
numframes2pick: 5

    # Plotting configuration
skeleton:
- - head
  - center
- - center
  - back
- - flowright
  - flowcenter
- - flowcenter
  - flowleft
skeleton_color: black
pcutoff: 0.6
dotsize: 12
alphavalue: 0.7
colormap: rainbow

    # Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 0
default_net_type: dlcrnet_ms5
default_augmenter: multi-animal-imgaug
default_track_method: ellipse
snapshotindex: -1
batch_size: 8

    # Cropping Parameters (for analysis and outlier frame detection)
cropping: false
    #if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624

    # Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by MMathisLab** [comment]
it looks like you are using a multi-animal project; so you have 18 key points per animal, and thus 36 (vs. the 18) error. DLC triangulate does not use the CSV files, we knly use the H5 files internal to ANY function in our code base. Are both your cameras using the same multi-animal model?

**Comment by bazigo76** [comment]
Oh I see ! 
Both of my camera use the same multi-animal model

**Comment by bazigo76** [comment]
in the context of my study, I had to follow 2 objects: my bee and my robotic flower (which therefore moves). To date, I have not been able to launch the deeplabcut 3d procedure in multi tracking. I tried this time in single animal, and it works. I selected the points of my bee, and the points of my flower. With a skeleton formed for my flower and for my bee respectively. I guess it's not very rigorous to do it this way. Why does it work in single and not in multi?
